backends/cxxrtl/cxxrtl.h

```diff
@@ -450,12 +450,18 @@ struct value : public expr_base<value<Bits>> {
 	std::pair<value<Bits>, bool /*CarryOut*/> alu(const value<Bits> &other) const {
 		value<Bits> result;
 		bool carry = CarryIn;
-		for (size_t n = 0; n < result.chunks; n++) {
+		// Handle full chunks first
+		for (size_t n = 0; n < result.chunks - 1; n++) {
 			result.data[n] = data[n] + (Invert ? ~other.data[n] : other.data[n]) + carry;
 			carry = (result.data[n] <  data[n]) ||
 			        (result.data[n] == data[n] && carry);
 		}
-		result.data[result.chunks - 1] &= result.msb_mask;
+		// Handle last chunk (mask before updating carry)
+		constexpr size_t last = result.chunks - 1;
+		result.data[last] = data[last] + (Invert ? ~other.data[last] : other.data[last]) + carry;
+		result.data[last] &= result.msb_mask;
```

Comment:
I'm pretty sure you can just use a conditional instead of duplicating all this code, this loop gets unrolled anyways.

Reply:
Ok, i wasn't sure if I could rely on it always being unrolled and optimised away.

Reply:
There isn't any pragmas to force it to unroll the loop, but in general, the performance goals of CXXRTL can only be achieved with a C++ compiler that aggressively unrolls loops with a small amount of conditionals in them. (There's already ternaries in all the loops!) So it's fine to do this here, too.